src/Octokit.Webhooks/WebhookEventProcessor.cs

```diff
@@ -65,1324 +66,1029 @@ public virtual void ProcessWebhook(IDictionary<string, StringValues> headers, st
-                    this.ProcessWatchWebhook(headers, body);
-                    break;
+                    return this.ProcessWatchWebhook(headers, body);
                 case WebhookEventType.WorkflowDispatch:
-                    this.ProcessWorkflowDispatchWebhook(headers, JsonSerializer.Deserialize<WorkflowDispatchEvent>(body)!);
-                    break;
+                    return this.ProcessWorkflowDispatchWebhook(headers, JsonSerializer.Deserialize<WorkflowDispatchEvent>(body)!);
                 case WebhookEventType.WorkflowJob:
-                    this.ProcessWorkflowJobWebhook(headers, body);
-                    break;
+                    return this.ProcessWorkflowJobWebhook(headers, body);
                 case WebhookEventType.WorkflowRun:
-                    this.ProcessWorkflowRunWebhook(headers, body);
-                    break;
+                    return this.ProcessWorkflowRunWebhook(headers, body);
             }
+
+            return Task.CompletedTask;
         }
 
-        private void ProcessBranchProtectionRuleWebhook(WebhookHeaders headers, string body)
+        private Task ProcessBranchProtectionRuleWebhook(WebhookHeaders headers, string body)
         {
             var branchProtectionRuleEvent = JsonSerializer.Deserialize<BranchProtectionRuleEvent>(body)!;
             switch (branchProtectionRuleEvent.Action)
             {
                 case BranchProtectionRuleActionValue.Created:
-                    this.ProcessBranchProtectionRuleWebhook(headers, branchProtectionRuleEvent, BranchProtectionRuleAction.Created);
-                    break;
+                    return this.ProcessBranchProtectionRuleWebhook(headers, branchProtectionRuleEvent, BranchProtectionRuleAction.Created);
                 case BranchProtectionRuleActionValue.Deleted:
-                    this.ProcessBranchProtectionRuleWebhook(headers, branchProtectionRuleEvent, BranchProtectionRuleAction.Deleted);
-                    break;
+                    return this.ProcessBranchProtectionRuleWebhook(headers, branchProtectionRuleEvent, BranchProtectionRuleAction.Deleted);
                 case BranchProtectionRuleActionValue.Edited:
-                    this.ProcessBranchProtectionRuleWebhook(headers, branchProtectionRuleEvent, BranchProtectionRuleAction.Edited);
-                    break;
+                    return this.ProcessBranchProtectionRuleWebhook(headers, branchProtectionRuleEvent, BranchProtectionRuleAction.Edited);
             }
+            return Task.CompletedTask;
         }
 
         [PublicAPI]
-        protected virtual void ProcessBranchProtectionRuleWebhook(WebhookHeaders headers, BranchProtectionRuleEvent branchProtectionRuleEvent, BranchProtectionRuleAction action)
-        {
-        }
+        protected virtual Task ProcessBranchProtectionRuleWebhook(WebhookHeaders headers, BranchProtectionRuleEvent branchProtectionRuleEvent, BranchProtectionRuleAction action) => Task.CompletedTask;
```

Comment:
Can you add `CancellationToken ct = default` all the way down?

Reply:
Just waiting for your response on this https://github.com/octokit/webhooks.net/pull/17#discussion_r748999146 before adding the cancellation token changes